src/datadog/random.cpp

```diff
@@ -0,0 +1,91 @@
+#include "random.h"
+
+#include <bitset>
+#include <random>
+
+#include "hex.h"
+#include "platform_util.h"
+
+namespace datadog {
+namespace tracing {
+namespace {
+
+extern "C" void on_fork();
+
+class Uint64Generator {
+  std::mt19937_64 generator_;
+  // The distribution used is for a _signed_ integer, and the default minimum
+  // value is zero.
+  // This means that we generate IDs with non-negative values that will always
+  // fit into an `int64_t`, which is a polite thing to do when you work with
+  // people who write Java.
+  std::uniform_int_distribution<std::int64_t> distribution_;
```

Comment:
That's a good point. Also, I think that the "produce 63 bits instead of 64 because uint64 is hard" idea is no longer relevant. We could just use 64 bits everywhere. What do you think?